### ISSUE TITLE
Code cleanup

### ISSUE THREAD

**Issue by muodov** [muodov/kociemba#17]
Since kociemba initially was a direct translation from java, the code inherited some awkward places in design. For example, in some places python list elements are iterated by index. Also, decomposition might be improved. 

These need to be fixed to keep the project healthy.

**Comment by ansiwen** [comment]
Also the C api doesn't look very consistent... I personally would prefer a C++ library anyways, with an python wrapper around it. I guess in the midterm keeping two libraries in sync will not be feasible.

**Comment by muodov** [comment]
Yep, I agree, C implementation definitely needs some care, at the moment it is more like a working PoC than a proper library. And C implementation is surely more important since solving the cube is performance-bound. Pure Python version might be deprecated in future, but it is working now and there are some cases when it is handy.
